CMakeLists.txt

```diff
@@ -53,25 +61,23 @@ endif()
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
+if (GPRT_BUILD_SHARED)
+  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
Is there a way we could make this specific to the GPRT target? I find that global compiler flags like this tend to break other submodules in a global project.


I think we should be using an optional SHARED keyword on the GPRT target rather than setting a global flag. https://cmake.org/cmake/help/latest/prop_tgt/POSITION_INDEPENDENT_CODE.html


Or this 
```
add_library(lib1 lib1.cpp)
set_property(TARGET lib1 PROPERTY POSITION_INDEPENDENT_CODE ON)
```